src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -810,6 +811,17 @@ public IMethodSymbol GetEntryPoint(CancellationToken cancellationToken)
             return (INamedTypeSymbol?)CommonGetSpecialType(specialType)?.GetITypeSymbol();
         }
 
+        public INamedTypeSymbol GetSpecialTypeOrThrow(SpecialType specialType)
```

Comment:
Does this need to be `public`?

Reply:
@cston This is used in the IDE layer so it needs to be public

Reply:
@333fred though for visibility since it's an API change

---
In reply to: [359999354](https://github.com/dotnet/roslyn/pull/40493#discussion_r359999354) [](ancestors = 359999354)

Reply:
If you don't want to make this public, you could add an extension method with the same name to the Workspaces layer and it will be visible throughout the IDE layers.

Reply:
It seems like a method we'd use in both layers. Is there an existing pattern for where to put extension methods like this that get shared as source between the two layers?

Reply:
Would we actually use this in the IDE? For us we have to be resilient to the design time build going sideways, so a lot of code can't use this.

Reply:
@jasonmalinowski yes, this PR contains one use of the method and one other place where someone, named jason, suggested we use this helper 😉

Reply:
It's in the fine print of the EULA tagging me in a review: I'm entitled to change my opinion for any reason, or for no reason whatsoever.

More seriously: if you want to expose this I guess we'll use it in that one spot. If you don't want to expose it...we'll be fine.

Reply:
I decided for the time to make this internal + use via shared source in the IDE. 

---
In reply to: [360019054](https://github.com/dotnet/roslyn/pull/40493#discussion_r360019054) [](ancestors = 360019054)